src/utils/editor/index.js

```diff
@@ -87,11 +87,37 @@ export function scrollToColumn(codeMirror: any, line: number, column: number) {
     "local"
   );
 
-  const scroller = codeMirror.getScrollerElement();
-  const centeredX = Math.max(left - scroller.offsetWidth / 2, 0);
-  const centeredY = Math.max(top - scroller.offsetHeight / 2, 0);
+  if (!isVisible(codeMirror, top, left)) {
+    const scroller = codeMirror.getScrollerElement();
+    const centeredX = Math.max(left - scroller.offsetWidth / 2, 0);
+    const centeredY = Math.max(top - scroller.offsetHeight / 2, 0);
 
-  codeMirror.scrollTo(centeredX, centeredY);
+    codeMirror.scrollTo(centeredX, centeredY);
+  }
+}
+
+function isVisible(codeMirror: any, top: number, left: number) {
+  function withinBounds(x, min, max) {
+    return x >= min && x <= max;
+  }
+
+  const scrollArea = codeMirror.getScrollInfo();
+
+  const charWidth = codeMirror.defaultCharWidth();
+  const inXView = withinBounds(
+    left,
+    scrollArea.left,
+    scrollArea.left + scrollArea.clientWidth - charWidth
+  );
+
+  const fontHeight = codeMirror.defaultTextHeight();
+  const inYView = withinBounds(
+    top,
+    scrollArea.top,
+    scrollArea.top + scrollArea.clientHeight - fontHeight
+  );
```

Comment:
how does this work?

Reply:
I did this just because if your scroll view area was not exactly a multiple of the font height then the last line may be cut off when you go to debug the last line, as it will still stay in view. So instead I said that if the top is below the top of the last lines fontHeight just realign and then you can see the whole line.